### PR TITLE
Check matrix conditioning

### PR DESCRIPTION
**Pull request by alexfikl**
The condition number for `0.5 * u + D(u, "avg")` seems to be huge when constructing the matrix with `pytential.symbolic.execute.build_matrix`, but it's not for `-0.5 * u + D(u, "avg")`.

@inducer

### CODE REVIEW

**Comment by alexfikl** [comment]
Doing and SVD on the two gave this: `side=-1` (left) and `side=+1` (right)

<p align="center">


</p>

It does seem like it has a singular value dangerously close to 0 there. This seems to stay there when going from a circle to an ellipse, so it doesn't seem like it's a fluke (?).

**Comment by inducer** [comment]
Oh right. Exterior Dirichlet Laplace has a nullspace, cf. [here](https://github.com/inducer/pytential/blob/094b75ba3ece5314b6ce6f7930d946fe5efbb8bb/pytential/symbolic/pde/scalar.py#L124-L127). Sorry, I should've remembered that when we chatted. Once you nudge the constant out of the nullspace (like [here](https://github.com/inducer/pytential/blob/094b75ba3ece5314b6ce6f7930d946fe5efbb8bb/pytential/symbolic/pde/scalar.py#L157-L167)), the conditioning should become entirely benign. So I don't think there's anything to see here, this is entirely expected.

**Comment by inducer** [comment]
(It might not hurt to keep that test around though.)

**Comment by alexfikl** [comment]
> (It might not hurt to keep that test around though.)

Yeah, sure. What should it test though? That there's a singular value close to zero in that one case?

**Comment by inducer** [comment]
Just slap a reasonable upper bound on it? Basically... not 1e10? :)

**Comment by alexfikl** [comment]
> Just slap a reasonable upper bound on it? Basically... not 1e10? :)

@inducer Added an assert like that and everything seems to pass, so it's good to go from my side.

File: test/test_matrix.py [hunk 1]
```diff
@@ -179,6 +179,96 @@ def test_build_matrix(ctx_factory, k, curve_fn, op_type, visualize=False):
...
+        ))
+
+    kappa = la.cond(mat)
+    _, sigma, _ = la.svd(mat)
+
+    logger.info("cond: %.5e sigma_max %.5e", kappa, sigma[0])
+
+    # NOTE: exterior Laplace has a nullspace
+    if side == +1 and op_type == "double":
+        assert kappa > 1.0e+9
```
Comment (inducer) [inline test/test_matrix.py:241]: Oh, sorry! What I meant was that we should test that the matrix with "ones(n, n)" added to it should have benign conditioning, as in the IE test. (This removes the nullspace.)

Comment (alexfikl) [reply test/test_matrix.py:246]: Added in 7e78c22. Better?

Comment (inducer) [reply test/test_matrix.py:246]: Yup, thanks!

**Pull request merged**